modules/s3/basic/main.tf

```
@@ -1,5 +1,10 @@
+locals {
+  # Name restrictions https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html
+  default_name = substr("${var.prefix}${var.app_id}-${var.env_id}-${replace(var.res_id, ".", "-")}", 0, 63)
```

When I was testing it the string was trimmed just after `-` making it not valid. Trimming it would make it more resilient for such situations
```suggestion
  default_name = trimsuffix(substr("${var.prefix}${var.app_id}-${var.env_id}-${replace(var.res_id, ".", "-")}", 0, 63), "-")
```